bootstrap/secret/secure_test.go

```diff
@@ -0,0 +1,254 @@
+
+	tests := []struct {
+		Name        string
+		Path        string
+		Keys        []string
+		Config      TestConfig
+		Client      secrets.SecretClient
+		ExpectError bool
+	}{
+		{"Valid Secure", "redis", []string{"username", "password"}, TestConfig{}, mock, false},
+		{"Invalid Secure", "missing", []string{"username", "password"}, TestConfig{}, mock, true},
+		{"Invalid No Client", "redis", []string{"username", "password"}, TestConfig{}, nil, true},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.Name, func(t *testing.T) {
+			target := NewSecureProvider(tc.Config, logger.MockLogger{}, nil)
+			target.SetClient(tc.Client)
+			actual, err := target.GetSecrets(tc.Path, tc.Keys...)
+			if tc.ExpectError {
+				require.Error(t, err)
+				return
+			}
+
+			require.NoError(t, err)
+			assert.Equal(t, expected, actual)
+		})
+	}
+}
+
+func TestSecureProvider_GetSecrets_Cached(t *testing.T) {
+	expected := map[string]string{"username": "admin", "password": "sam123!"}
+
+	mock := &mocks.SecretClient{}
+	// Use the Once method so GetSecrets can be changed below
+	mock.On("GetSecrets", "redis", "username", "password").Return(expected, nil).Once()
+
+	target := NewSecureProvider(nil, logger.MockLogger{}, nil)
+	target.SetClient(mock)
+
+	actual, err := target.GetSecrets("redis", "username", "password")
+	require.NoError(t, err)
+	assert.Equal(t, expected, actual)
+
+	// Now have mock return error if it is called which should not happen of secrets are cached
+	mock.On("GetSecrets", "redis", "username", "password").Return(nil, errors.New("No Cached"))
+	actual, err = target.GetSecrets("redis", "username", "password")
+	require.NoError(t, err)
+	assert.Equal(t, expected, actual)
+
+	// Now check for error when not all requested keys not in cache.
+	mock.On("GetSecrets", "redis", "username", "password2").Return(nil, errors.New("No Cached"))
+	actual, err = target.GetSecrets("redis", "username", "password2")
+	require.Error(t, err)
+}
+
+func TestSecureProvider_GetSecrets_Cached_Invalidated(t *testing.T) {
+	expected := map[string]string{"username": "admin", "password": "sam123!"}
+
+	mock := &mocks.SecretClient{}
+	// Use the Once method so GetSecrets can be changed below
+	mock.On("GetSecrets", "redis", "username", "password").Return(expected, nil).Once()
+	mock.On("StoreSecrets", "redis", expected).Return(nil)
+
+	target := NewSecureProvider(nil, logger.MockLogger{}, nil)
+	target.SetClient(mock)
+
+	actual, err := target.GetSecrets("redis", "username", "password")
+	require.NoError(t, err)
+	assert.Equal(t, expected, actual)
+
+	// Invalidate the secrets cache by storing new secrets
+	err = target.StoreSecrets("redis", expected)
+	require.NoError(t, err)
+
+	// Now have mock return error is it is called which should now happen if the cache was properly invalidated by the above call to StoreSecrets
+	mock.On("GetSecrets", "redis", "username", "password").Return(nil, errors.New("No Cached"))
+	actual, err = target.GetSecrets("redis", "username", "password")
```

Comment:
don't need `actual` here since not test it.
```suggestion
	_, err = target.GetSecrets("redis", "username", "password")
```